temp_cache.rb

```diff
@@ -14,22 +30,31 @@ def create!
     Dir.mkdir(cache_path)
   end
 
-  def check_file_age
+  def check_file_age!
```

Comment:
Weird convention choice, it has side-effects of possibly deleting files (which it's soley responsible for) but also might throw an error, but not from  app code. 

Future: -> Work on possible File System error handling